tests/serde-de.rs

```diff
@@ -110,48 +110,6 @@ fn nested_collection() {
     );
 }
 
-#[test]
-fn collection_of_enums() {
-    #[derive(Debug, Deserialize, PartialEq)]
-    enum MyEnum {
-        A(String),
-        B { name: String, flag: bool },
-        C,
-    }
-
-    #[derive(Debug, Deserialize, PartialEq)]
-    struct MyEnums {
-        // TODO: This should be #[serde(flatten)], but right now serde don't support flattening of sequences
-        // See https://github.com/serde-rs/serde/issues/1905
-        #[serde(rename = "$value")]
-        items: Vec<MyEnum>,
-    }
```

Comment:
This is a much more complex test than the one that replaces it (https://github.com/tafia/quick-xml/blob/d34b77972ccaad505c3348761382fe7d01ca49ce/tests/serde-de.rs#L1261=), is it really a full replacement?

Reply:
Actually, yes. The mapping should be independent and composable, so we shouldn't bother of internals of the `MyEnum` type (or `Item` type in `Project`). For now we don't have a tests that ensures that mapping is really additive in all cases (which is why #435 is happened), but
- that check should be added explicitly and for all combinations
- in that test mapping is independent and composable

I'll plan to add necessary tests that would ensure that mapping is always composable (may be in form of doctests in #369)